nipype/workflows/dmri/fsl/artifacts.py

```diff
@@ -545,66 +550,93 @@ def sdc_fmb(name='fmb_correction',
-                     args='-kernel sphere 5 -dilM'), name='MskDilate')
-    pha2rads = pe.Node(niu.Function(input_names=['in_file'], output_names=['out_file'],
-                       function=siemens2rads), name='PreparePhase')
+    dilate = pe.Node(fsl.maths.MathsCommand(
+        nan2zeros=True, args='-kernel sphere 5 -dilM'), name='MskDilate')
+    pha2rads = pe.Node(niu.Function(
+        input_names=['in_file'], output_names=['out_file'],
+        function=siemens2rads), name='PreparePhase')
     prelude = pe.Node(fsl.PRELUDE(process3d=True), name='PhaseUnwrap')
-    rad2rsec = pe.Node(niu.Function(input_names=['in_file', 'delta_te'],
-                       output_names=['out_file'], function=rads2radsec), name='ToRadSec')
-    rad2rsec.inputs.delta_te = bmap_params['delta_te']
-
-    avg_b0 = pe.Node(niu.Function(input_names=['in_dwi', 'in_bval'],
-                     output_names=['out_file'], function=b0_average), name='b0_avg')
-
-    flirt = pe.Node(fsl.FLIRT(interp='spline', cost='normmi', cost_func='normmi',
-                    dof=6, bins=64, save_log=True, padding_size=10,
-                    searchr_x=[-4, 4], searchr_y=[-4, 4], searchr_z=[-4, 4],
-                    fine_search=1, coarse_search=10),
-                    name='BmapMag2B0')
-    applyxfm = pe.Node(fsl.ApplyXfm(interp='spline', padding_size=10, apply_xfm=True),
-                       name='BmapPha2B0')
+    rad2rsec = pe.Node(niu.Function(
+        input_names=['in_file', 'delta_te'], output_names=['out_file'],
+        function=rads2radsec), name='ToRadSec')
+
+    avg_b0 = pe.Node(niu.Function(
+        input_names=['in_dwi', 'in_bval'], output_names=['out_file'],
+        function=b0_average), name='b0_avg')
+
+    fmm2b0 = pe.Node(ants.Registration(output_warped_image=True),
+                     name="FMm_to_B0")
+    fmm2b0.inputs.transforms = ['Rigid'] * 2
+    fmm2b0.inputs.transform_parameters = [(1.0,)] * 2
+    fmm2b0.inputs.number_of_iterations = [[50], [20]]
+    fmm2b0.inputs.dimension = 3
+    fmm2b0.inputs.metric = ['Mattes', 'Mattes']
+    fmm2b0.inputs.metric_weight = [1.0] * 2
+    fmm2b0.inputs.radius_or_number_of_bins = [64, 64]
+    fmm2b0.inputs.sampling_strategy = ['Regular', 'Random']
+    fmm2b0.inputs.sampling_percentage = [None, 0.2]
+    fmm2b0.inputs.convergence_threshold = [1.e-5, 1.e-8]
+    fmm2b0.inputs.convergence_window_size = [20, 10]
+    fmm2b0.inputs.smoothing_sigmas = [[6.0], [2.0]]
+    fmm2b0.inputs.sigma_units = ['vox'] * 2
+    fmm2b0.inputs.shrink_factors = [[6], [1]]  # ,[1] ]
+    fmm2b0.inputs.use_estimate_learning_rate_once = [True] * 2
+    fmm2b0.inputs.use_histogram_matching = [True] * 2
+    fmm2b0.inputs.initial_moving_transform_com = 0
+    fmm2b0.inputs.collapse_output_transforms = True
+    fmm2b0.inputs.winsorize_upper_quantile = 0.995
+
+    applyxfm = pe.Node(ants.ApplyTransforms(
+        dimension=3, interpolation='BSpline'), name='FMp_to_B0')
```

Comment:
`BSpline` gave me a whole lot of trouble on non-negative variance images (and appears to be an ITK issue). so make sure that this works here. otherwise you may want to go with `Linear`. it will make the field smoother, but may not be that big an issue here.